coderd/insights.go

```diff
@@ -43,3 +51,329 @@ func (api *API) deploymentDAUs(rw http.ResponseWriter, r *http.Request) {
 	}
 	httpapi.Write(ctx, rw, http.StatusOK, resp)
 }
+
+// @Summary Get insights about user latency
+// @ID get-insights-about-user-latency
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Insights
+// @Success 200 {object} codersdk.UserLatencyInsightsResponse
+// @Router /insights/user-latency [get]
+func (api *API) insightsUserLatency(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	if !api.Authorize(r, rbac.ActionRead, rbac.ResourceDeploymentValues) {
+		httpapi.Forbidden(rw)
+		return
+	}
+
+	// TODO(mafredri): Client or deployment timezone?
+	// Example:
+	// - I want data from Monday - Friday
+	// - I'm UTC+3 and the deployment is UTC+0
+	// - Do we select Monday - Friday in UTC+0 or UTC+3?
```

Comment:
Friday UTC+3, using timezone from the incoming request. Coder admins won't necessarily know or care about the localtime of the server on which the deployment is running.

Reply:
One factor here is that two people looking at "the same data" may see different numbers. They may share screen captures and wonder why the data doesn't add up. An anchor like deployment TZ would help in that case.

Reply:
Yeah, showing the user's relative difference from the deployment TZ (or even just showing the GMT offset in the graph) would help indicate this.